frontend/src/metabase/dashboard/utils.ts

```diff
@@ -97,7 +96,8 @@ export function isLinkDashCard(dashcard: DashboardCard) {
 }
 
 export function isNativeDashCard(dashcard: DashboardCard) {
-  return dashcard.card && new Question(dashcard.card).isNative();
+  // The `dataset_query` is null for questions on a dashboard the user doesn't have access to
```

Comment:
Please update `UnsavedCard["dataset_query"]` type to account for the `null`.
This comment could live next to that property, not here.

Reply:
Do you mind if I do that in a separate PR?
Updating the type will break type check in many places. It will pollute this diff.